packages/cozy-pouch-link/src/CozyPouchLink.js

```diff
@@ -442,19 +446,42 @@ class PouchLink extends CozyLink {
     return parseMutationResult(mutation.document, res)
   }
 
+  async updateDocuments(mutation) {
+    const docs = mutation.documents
+    const res = await this.dbMethod('bulkDocs', mutation)
+    const badResults = res.filter(x => !x.ok)
+    if (badResults.length > 0) {
+      console.warn('Error while bulk saving, bad results', badResults)
+      throw new Error('Error while bulk saving, see above for errors')
+    }
```

Comment:
I don't know if we should throw here. Since some documents can have been already updated in the DB we can have difference between the DB and the CozyClient.

Reply:
What do you suggest ? I think in this case, the app should refetch all affected documents.

Reply:
How can the app can know what documents to refetch? We don't return them since we throw, no? 

Let's say the doc1 updates is OK, doc2 is KO.

Why the current implementation I don't see how the app can handle the case. 

```js 
try {
const data1 = {id: 1}
const doc1 = {...data1, foo:'bar'}
const doc2 = {...data2, foo:'bar'}
const updatedDocs = await Cozyclient.saveAll([doc1, doc2])
} catch(e){
const testdoc1 = await CozyClient.query(id1) // => contain foo:bar 
const testdocs2 = await CozyClient.query(id2) // => contain no foo:bar  
}
```
What do you think of returning something like this:

```js
const updatedDocs = await Cozyclient.saveAll([doc1, doc2, doc3])
if(updatedDocs.errored){
   const failedUpdates = e.errored
   failedUpdates.map(doc => {
     console.log(doc.reason)
     console.log(doc.error)
  }
}
if(updatedDocs.data){
  const successfulUpdate = updatedDocs.data....
}
```

So instead of throwing an error and not returning the result, we return the result and the errors. The return is a bit different than our others methods since `errored` attribute doesn't exist in our `normalization` but I think it can be a good move. WDYT?

Reply:
I went a different route in https://github.com/cozy/cozy-client/pull/995/commits/dfc32c003ed739b0c81b21e93839280f1b01a7b6 , I attached the bad docs to the error. WDYT ?

Reply:
I think that if I went the way of the object with the `errored` object returned as part of th result, I'd need to make more changes to the `mutate` function from cozy-client, which only rely on exceptions to dispatch a receiveMutationError.

For now, inside cozy-client we are not accustomed to checking an `errored` field, it would be a new behavior and I think it is better to stay consistent.

Reply:
I've extracted BulkEditError inside cozy-client (remove duplication of the custom error class) and added methods on the error to get the updated documents as well as the errors.

See [`8e5aca5` (#995)](https://github.com/cozy/cozy-client/pull/995/commits/8e5aca5bfde7c0c3675a95ceb638ba37e820aa32)